junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/JupiterEngineDescriptor.java

```diff
@@ -31,7 +32,9 @@ public JupiterEngineDescriptor(UniqueId uniqueId) {
 
 	@Override
 	public JupiterEngineExecutionContext before(JupiterEngineExecutionContext context) {
-		return context.extend().withExtensionRegistry(createRegistryWithDefaultExtensions()).build();
+		boolean autoDetect = Boolean.parseBoolean(
+			context.getConfigurationParameters().get(EXTENSIONS_AUTODETECT_PROPERTY_NAME).orElse("false"));
+		return context.extend().withExtensionRegistry(createRegistryWithDefaultExtensions(autoDetect)).build();
```

Comment:
Ahhhh. I see you came up with an alternative solution for accessing the `ConfigurationParameters`. That also works!

I'll ponder which is better to keep in place, but for now this is sufficient.